test/test-fs.c

```diff
@@ -2740,3 +2740,55 @@ TEST_IMPL(fs_read_write_null_arguments) {
 
   return 0;
 }
+
+
+TEST_IMPL(get_osfhandle_read_write) {
+  uv_buf_t iovs[1];
+  int r;
+  uv_os_fd_t fd;
+
+  /* Setup. */
+  unlink("test_file");
+
+  loop = uv_default_loop();
+
+  r = uv_fs_open(NULL,
+                 &open_req1,
+                 "test_file",
+                 O_RDWR | O_CREAT,
+                 S_IWUSR | S_IRUSR,
+                 NULL);
+  ASSERT(r >= 0);
+  ASSERT(open_req1.result >= 0);
+  uv_fs_req_cleanup(&open_req1);
+
+  iovs[0] = uv_buf_init(test_buf, sizeof(test_buf));
+  r = uv_fs_write(NULL, &write_req, open_req1.result, iovs, 1, 0, NULL);
+  ASSERT(r >= 0);
+  ASSERT(write_req.result >= 0);
+  uv_fs_req_cleanup(&write_req);
+
+  fd = uv_get_osfhandle(open_req1.result);
+#ifdef _WIN32
+  ASSERT(fd != INVALID_HANDLE_VALUE);
+#else
+  ASSERT(fd >= 0);
+#endif
+
+  iovs[0] = uv_buf_init(buf, sizeof(test_buf));
+  r = uv_fs_read(NULL, &read_req, fd, iovs, 1, 0, NULL);
```

Comment:
Hm, yeah, I just realized that, in which case my whole test case doesn't make sense anymore. Can you think of any uv function that interacts with files that could accept a `uv_os_fd_t`, to ensure its validity? Otherwise, I can reduce the test case to just check that the fd is greater than zero on UNIX, and not an invalid handle on Windows. What do you think?